packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx

```diff
@@ -1,44 +1,128 @@
 import React, { FC } from 'react';
-import { ColorPalette, IPostForms } from '@vse-bude/shared';
+import { Asset } from 'react-native-image-picker';
+import {
+  FullUserProfileDto,
+  ICreatePost,
+  ProductStatus,
+} from '@vse-bude/shared';
 import {
   DropDown,
   Input,
-  InfoIcon,
   Text,
-  TouchableOpacity,
   View,
-  Popover,
+  SecondaryButton,
+  PrimaryButton,
 } from '~/components/components';
-import { useAppForm, useAppSelector, useTranslation } from '~/hooks/hooks';
-import { CALLING_CODE, CITIES, COUNTRIES, CURRENCY } from '~/mock/new-item';
+import {
+  useAppForm,
+  useAppSelector,
+  useTranslation,
+  useCustomTheme,
+  useState,
+  useAppDispatch,
+  useEffect,
+  useNavigation,
+} from '~/hooks/hooks';
+import { products as productsActions } from '~/store/actions';
+import { CONDITION } from '~/mock/new-item';
 import { globalStyles } from '~/styles/styles';
-import { selectCategories } from '~/store/categories/selectors';
 import { categoryForDropdown } from '~/helpers/category/format-category-for-dropdown';
+import { ButtonsContainer } from '~/screens/components/components';
+import { ButtonAppearance, DataStatus } from '~/common/enums/enums';
+import { selectCategories, selectDataStatusProducts } from '~/store/selectors';
+import { productsPostSchema } from '~/validation-schemas/validation-schemas';
+import { notification } from '~/services/services';
+import { makePostParser } from '~/helpers/helpers';
 import { AddPhotos } from '../add-photos/add-photos';
-
 import { useStyles } from './styles';
 
-const NewItemForm: FC = () => {
+type Props = {
+  personalInfo: FullUserProfileDto;
+};
+
+const NewItemForm: FC<Props> = ({ personalInfo }) => {
   const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+  const navigation = useNavigation();
+  const { colors } = useCustomTheme();
   const styles = useStyles();
   const categories = useAppSelector(selectCategories);
+  const dataStatusProducts = useAppSelector(selectDataStatusProducts);
+  const isLoading = dataStatusProducts === DataStatus.PENDING;
   const formattedCategories =
     categories && categories.length ? categoryForDropdown(categories) : null;
 
-  const { control, errors } = useAppForm<IPostForms>({
+  const { control, errors, handleSubmit } = useAppForm<ICreatePost>({
     defaultValues: {
-      currency: 'UAH',
-      country: 'Ukraine',
-      callingCode: 'UA',
+      category: '',
+      title: '',
+      description: '',
+      condition: '',
+      currency: t('common:currency.UAH'),
+      price: 0,
+      country: personalInfo.userAddress?.country || '',
+      city: personalInfo.userAddress?.city || '',
+      phone: personalInfo.phone?.slice(4) || '',
```

Comment:
What if it's stored as "+38 050 123 45 67"?

Reply:
corrected